matchmake-extension/auto_matchmake_with_param_postpone.go

```diff
@@ -0,0 +1,85 @@
+package matchmake_extension
+
+import (
+	nex "github.com/PretendoNetwork/nex-go"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	match_making_types "github.com/PretendoNetwork/nex-protocols-go/match-making/types"
+	matchmake_extension "github.com/PretendoNetwork/nex-protocols-go/matchmake-extension"
+)
+
+func autoMatchmakeWithParam_Postpone(err error, client *nex.Client, callID uint32, autoMatchmakeParam *match_making_types.AutoMatchmakeParam) uint32 {
+	if commonMatchmakeExtensionProtocol.cleanupSearchMatchmakeSessionHandler == nil {
+		logger.Warning("MatchmakeExtension::AutoMatchmake_Postpone missing CleanupSearchMatchmakeSessionHandler!")
+		return nex.Errors.Core.NotImplemented
+	}
+
+	if err != nil {
+		logger.Error(err.Error())
+		return nex.Errors.Core.InvalidArgument
+	}
+
+	server := commonMatchmakeExtensionProtocol.server
+
+	// A client may disconnect from a session without leaving reliably,
+	// so let's make sure the client is removed from the session
+	common_globals.RemoveClientFromAllSessions(client)
+
+	var matchmakeSession *match_making_types.MatchmakeSession
+	matchmakeSession = autoMatchmakeParam.SourceMatchmakeSession
+
+	searchMatchmakeSession := matchmakeSession.Copy().(*match_making_types.MatchmakeSession)
+	commonMatchmakeExtensionProtocol.cleanupSearchMatchmakeSessionHandler(searchMatchmakeSession)
+	sessionIndex := common_globals.FindSessionByMatchmakeSession(searchMatchmakeSession)
```

Comment:
`AutoMatchmakeWithParam_Postpone` has search criteria, so I would find the session using `FindSessionsByMatchmakeSessionSearchCriterias`